tests/integration/targets/ec2_spot_instance/tasks/main.yaml

```diff
@@ -221,14 +271,26 @@
           filters:
             vpc-id: "{{ vpc_result.vpc.id }}"
 
+      - name: get all spot requests created during test
+        ec2_spot_instance_info:
+          filters:
+            tag:ansible-test: "{{ resource_prefix }}"
+        register: spot_request_list
+
       - name: remove spot instance requests
         ec2_spot_instance:
           spot_instance_request_ids:
-            - '{{ create_result.spot_request.spot_instance_request_id }}'
-            - '{{ complex_create_result.spot_request.spot_instance_request_id }}'
+            - '{{ item.spot_instance_request_id }}'
           state: 'absent'
         ignore_errors: true
         retries: 5
+        with_items: "{{ spot_request_list.spot_request }}"
+
+      - name: Assert that the created spot instance requests are removed
```

Comment:
Last small thing then this is ready to merge!

We don't actually want to have this assertion.  It happens to be passing because deleting the instances has the effect of cancelling the spot requests, but this doesn't actually test the result of the removal task because the var is coming from the info task.   We generally don't need assertions in the cleanup block so we can just remove this task.